okhttp/src/jvmMain/kotlin/okhttp3/internal/http2/Http2Connection.kt

```diff
@@ -237,6 +237,11 @@ class Http2Connection internal constructor(builder: Builder) : Closeable {
     val stream: Http2Stream
     val streamId: Int
 
+    // trigger interrupts before we write to the buffered output
+    // since flush is likely where we get notified.
+    if (Thread.interrupted())
+      throw InterruptedIOException()
```

Comment:
Should we reset this?

Reply:
Can change to 

```
    if (Thread.currentThread().isInterrupted)
      throw InterruptedIOException()
```

Reply:
Is there a test that would differentiate this? I suspect it’s probably not necessary as we’re not doing any blocking calls between here and flush().

Reply:
An interrupted thread should be in one of these states:

 1. Doing non-blocking work with the `Thread.interrupted()` bit set to true
 2. Throwing an InterruptedException or InterruptedIOException

Your code the idiomatic way to go from 1 to 2.

Reply:
I don't think it was the flush here I was worried about.  It was all the stream accounting work between here and that flush.

I wanted to minimise the race window, but failing before we create the stream.

Reply:
> Is there a test that would differentiate this? I suspect it’s probably not necessary as we’re not doing any blocking calls between here and flush().

I'll see how to create a test here.